src/coreclr/inc/winwrap.h

```diff
@@ -37,144 +23,10 @@
-#undef GetFileAttributes
-#undef GetFileAttributesEx
-#undef FindFirstFileEx
-#undef FindFirstFile
-#undef FindNextFile
-#undef CopyFile
-#undef CopyFileEx
-#undef MoveFile
-#undef CreateHardLink
-#undef CreateNamedPipe
-#undef WaitNamedPipe
-#undef LookupPrivilegeValue
-#undef GetVersionEx
-
-// winuser.h
-#undef MAKEINTRESOURCE
-#undef GetUserObjectInformation
-#undef GetMessage
-
-#undef SendMessage
-#undef CharLower
-#undef MessageBox
-#undef GetClassName
-#undef LoadString
-#undef GetCalendarInfo
-#undef GetDateFormat
-#undef GetTimeFormat
-#undef LCMapString
-
-#endif // !defined(__TODO_PORT_TO_WRAPPERS__)
-
-//
-// NT supports the wide entry points.  So we redefine the wrappers right back
-// to the *W entry points as macros.  This way no client code needs a wrapper on NT.
-//
-
-// winbase.h
-#define WszFormatMessage   FormatMessageW
-#define WszCreateMutex CreateMutexW
-#define WszOpenMutex OpenMutexW
-#define WszCreateEvent CreateEventW
-#define WszOpenEvent OpenEventW
-#define WszCreateWaitableTimer CreateWaitableTimerW
-#define WszCreateFileMapping CreateFileMappingW
-#define WszGetModuleHandle GetModuleHandleW
-#define WszGetModuleHandleEx GetModuleHandleExW
-#define WszGetCommandLine GetCommandLineW
-#define WszSetEnvironmentVariable SetEnvironmentVariableW
-#define WszExpandEnvironmentStrings ExpandEnvironmentStringsW
-#define WszOutputDebugString OutputDebugStringW
-#define WszFindResource FindResourceW
-#define WszFindResourceEx FindResourceExW
-#define WszBeginUpdateResource BeginUpdateResourceW
-#define WszUpdateResource UpdateResourceW
-#define WszEndUpdateResource EndUpdateResourceW
-#define WszGetPrivateProfileInt GetPrivateProfileIntW
-#define WszGetSystemDirectory GetSystemDirectoryW
-#define WszCreateNamedPipe CreateNamedPipeW
-#define WszWaitNamedPipe WaitNamedPipeW
-#define WszLookupPrivilegeValue LookupPrivilegeValueW
-
-// winuser.h
-#define WszMAKEINTRESOURCE MAKEINTRESOURCEW
-#define WszGetUserObjectInformation GetUserObjectInformationW
-#define WszGetMessage GetMessageW
-#define WszSendMessage SendMessageW
-#define WszCharLower CharLowerW
-#define WszGetClassName GetClassNameW
-#define WszLoadString LoadStringW
-#define WszRegOpenKeyEx RegOpenKeyExW
-#define WszRegOpenKey(hKey, wszSubKey, phkRes) RegOpenKeyExW(hKey, wszSubKey, 0, KEY_ALL_ACCESS, phkRes)
-#define WszRegQueryValue RegQueryValueW
-#define WszRegQueryValueEx RegQueryValueExW
-
-#define WszRegQueryInfoKey RegQueryInfoKeyW
-#define WszRegEnumValue RegEnumValueW
-#define WszRegEnumKeyEx RegEnumKeyExW
-#define WszGetCalendarInfo GetCalendarInfoW
-#define WszGetDateFormat GetDateFormatW
-#define WszGetTimeFormat GetTimeFormatW
-#define WszLCMapString LCMapStringW
-#define WszMultiByteToWideChar MultiByteToWideChar
-#define WszWideCharToMultiByte WideCharToMultiByte
-#define WszCreateSemaphore(_secattr, _count, _maxcount, _name) CreateSemaphoreExW((_secattr), (_count), (_maxcount), (_name), 0, MAXIMUM_ALLOWED | SYNCHRONIZE | SEMAPHORE_MODIFY_STATE)
```

Comment:
I am keeping the CreateSemaphore -> CreateSemaphoreEx forwarder to avoid behavior changes. I am not sure what it was for, it is probably working around a bug somewhere.

Reply:
I am a bit curious why it was forwarded, perhaps another artifact of the Win9x timeframe.  It seems that it is used in UTSemReadWrite which is used in the metadata code.  Quickly poking at some of the references it seems it is related in preventing collisions when creating new metadata entries into the metadata tables.

Reply:
I came from Windows Phone actually:
```
// CoreSystem has CreateSemaphoreExW but not CreateSemaphoreW. It's easy to map one to the other but
// unfortunately CreateSemaphoreExW wasn't introduced until Vista. We can use it under FEATURE_CORESYSTEM
// though (Win8 minimum OS version). We don't have the function in our WinSDK yet, so we'll need to pull in
// the extern manually.
```
I will get rid of CreateSemaphoreW  => CreateSemaphoreExW forwarder.

Reply:
Hmm, this has coupling with the PAL on non-Windows and use of PAL APIs in the BCL. It would not be an improvement get rid of this forwarder.